### PR TITLE
Fix `TestTopoCustomRule` flakiness

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This pull request tries to remove the flakiness of `TestTopoCustomRule`. This test was failing due to port already being used, to reduce this error the topology E2E tests are now split across several CI clusters.

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: test/config.json [hunk 1]
```diff
@@ -791,7 +791,7 @@
 			"Args": ["vitess.io/vitess/go/test/endtoend/topotest/zk2", "--topo-flavor=zk2"],
 			"Command": [],
 			"Manual": false,
-			"Shard": "25",
+			"Shard": "vtgate_topo_sk2",
```
Comment (deepthi) [inline test/config.json:794]: tiny typo: sk2 -> zk2

Comment (deepthi) [reply test/config.json:794]: Also seems like there should be a workflow file for this?

Comment (frouioui) [reply test/config.json:794]: sorry, I did not intend to do this change, I will revert it

Comment (frouioui) [reply test/config.json:794]: @deepthi, actually `Cluster 25` does not exist, is this normal?

**Review by deepthi — APPROVED** [review]

**Pull request merged**